includes/subscriptions/class-wc-payments-subscriptions-empty-state-manager.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Class WC_Payments_Subscriptions_Empty_State.
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class for loading WooCommerce Payments Subscription empty state screen.
+ */
+class WC_Payments_Subscriptions_Empty_State_Manager {
+
+	use WC_Payments_Subscriptions_Utilities;
+
+	/**
+	 * WC_Payments_Account instance to get information about the account.
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+
+	/**
+	 * WC_Payments_Subscriptions_Empty_State Constructor
+	 *
+	 * @param WC_Payments_Account $account Account class instance.
+	 */
+	public function __construct( WC_Payments_Account $account ) {
+		$this->account = $account;
+
+		if ( ! $this->is_subscriptions_plugin_active() ) {
+			add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts_and_styles' ] );
+			add_filter( 'woocommerce_subscriptions_not_found_label', [ $this, 'replace_subscriptions_empty_state' ] );
+		}
+	}
+
+	/**
+	 * Enqueues the WCPay Subscription empty state scripts and styles.
+	 */
+	public function enqueue_scripts_and_styles() {
+		$screen = get_current_screen();
+
+		// Only enqueue the scripts on the admin subscriptions screen.
+		if ( ! $screen || 'edit-shop_subscription' !== $screen->id ) {
+			return;
+		}
+
+		$script_src_url    = plugins_url( 'dist/subscriptions-empty-state.js', WCPAY_PLUGIN_FILE );
+		$script_asset_path = WCPAY_ABSPATH . 'dist/subscriptions-empty-state.asset.php';
+		$script_asset      = file_exists( $script_asset_path ) ? require_once $script_asset_path : [ 'dependencies' => [] ];
+		$wcpay_settings    = [
+			'connectUrl'    => WC_Payments_Account::get_connect_url(),
+			'isConnected'   => $this->account->is_stripe_connected(),
+			'newProductUrl' => esc_url( admin_url( 'post-new.php?post_type=product' ) ),
```

Comment:
In the ticket description I proposed using `post-new.php?post_type=product&select_subscription=true&subscription_pointers=true` as it seems like a nice user experience, especially if the merchant is creating their first subscriptions product.

I originally discovered this in subscriptions core → https://github.com/Automattic/woocommerce-subscriptions-core/blob/21d95a568dc3e33b4e364d4a26abb82da41f1857/includes/admin/class-wc-subscriptions-admin.php#L1258

Maybe we could even reuse this helper function directly? (it's public static)

Reply:
Oh neat. This also fixes the problem of it not auto-selecting the simple subscription product type. 👍